tests/e2e/framework/utils.go

```diff
@@ -384,6 +386,139 @@ func (f *Framework) AssertMetricsEndpointUsesHTTPVersion(endpoint, version strin
 	return nil
 }
 
+// createServiceAccount creates a service account
+func (f *Framework) SetupRBACForMetricsTest() error {
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return fmt.Errorf("Failed to find oc binary: %v", err)
+	}
+	cmd := exec.Command(ocPath,
+		"create", "sa", "prometheus-query-sa", "-n", f.OperatorNamespace)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to create service account: %v", err)
+	}
+
+	cmd = exec.Command(ocPath,
+		"adm", "policy", "add-cluster-role-to-user", "cluster-monitoring-view", "-z", "prometheus-query-sa", "-n", f.OperatorNamespace)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to add cluster role to user: %v", err)
+	}
+	return nil
+}
+
+// CleanupRBACForMetricsTest deletes the service account
+func (f *Framework) CleanUpRBACForMetricsTest() error {
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return fmt.Errorf("Failed to find oc binary: %v", err)
+	}
+	cmd := exec.Command(ocPath,
+		"delete", "sa", "prometheus-query-sa", "-n", f.OperatorNamespace)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("Failed to delete service account: %v", err)
+	}
+	return nil
+}
+func getCurlPrometheusCMD() string {
+	return `TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token) && { curl -k -s https://prometheus-k8s.openshift-monitoring.svc.cluster.local:9091/api/v1/targets --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt -H "Authorization: Bearer $TOKEN"; }`
+}
+
+func (f *Framework) GetPrometheusMetricTargets() ([]promv1.Target, error) {
+	var metricsTargets []promv1.Target
+
+	namespace := f.OperatorNamespace
+	ocPath, err := exec.LookPath("oc")
+	if err != nil {
+		return metricsTargets, fmt.Errorf("error getting oc path %s", err)
+	}
+	cmd := exec.Command(ocPath,
+		"run", "--rm", "-i", "--restart=Never", "--image=registry.fedoraproject.org/fedora:latest",
+		"-n", namespace, "--overrides={\"spec\": {\"serviceAccountName\": \"prometheus-query-sa\"}}", "metrics-test", "--", "bash", "-c",
+		getCurlPrometheusCMD(),
+	)
+	out, err := cmd.CombinedOutput()
```

Comment:
The output we're getting here is the metrics data, right?

Reply:
yes, that's correct